### PR TITLE
Fine-Tuning Scheduler Tutorial Update for Lightning/PyTorch 2.1.0

### PR DESCRIPTION
**Pull request by speediedan**
Validated the Fine-Tuning Scheduler (FTS) tutorial for FTS/Lightning/PyTorch `2.1.0` (as of the final PyTorch RC and the [latest `2.1.0` Lightning commit](https://github.com/Lightning-AI/lightning/tree/d1f8b0f7669aa808df5fcc2934455be3c83b4bae)). 

The only minor change in this PR is to remove an ephemeral yaml fine-tuning schedule. 

Note that due to an [apparent internal issue with tutorial publishing](https://github.com/Lightning-AI/tutorials/issues/238), the Fine-Tuning Scheduler 2.0 tutorial update was successfully committed but was never published. Since the aforementioned `papermill` job issue was unfortunately never isolated/resolved [^1], I'm hoping this PR can be successfully merged and the publish validated to ensure the Fine-Tuning Scheduler tutorial gets updated for the upcoming `2.1.0` release.

Thank you for all your work and your consistently valuable contributions to the open-source ML community!

## Before submitting

- [X] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [X] Did you make sure to **update the docs**?
- [X] Did you write any new **necessary tests**?

## What does this PR do?

Updates the Fine-Tuning Scheduler tutorial to better support `2.1.0` and prepare for `2.1.0` tutorial publishing.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

[^1]: In case the issue re-emerges (since the original Azure build logs containing the error have been purged) I should note that I could not reproduce the `git-lfs` error (in the `'Git check & switch branch'` step) of the `papermill` job in a local container but that the updated tutorial was ready to be published. Remotely failed `Git check & switch branch` working locally: ```
  root@57ded931c266:/fts_tutorial#     git fetch --all
  Fetching origin
  root@57ded931c266:/fts_tutorial#     git ls-remote --heads origin publication | grep publication >/dev/null
  root@57ded931c266:/fts_tutorial#     if [ "$?" == "1" ] ; then echo "Branch doesn't exist"; exit; fi
  root@57ded931c266:/fts_tutorial#     git checkout publication
  Already on 'publication'
  Your branch is up to date with 'origin/publication'.
  root@57ded931c266:/fts_tutorial#     git show-ref publication
  491d2cfc11259e19d9c8054e40844ca731840927 refs/heads/publication
  491d2cfc11259e19d9c8054e40844ca731840927 refs/remotes/origin/publication
  root@57ded931c266:/fts_tutorial#     git pull
  Already up to date.
  ```

### CODE REVIEW

**Comment by speediedan** [comment]
Marked "Ready for Review" despite the failing check as the check failure is due to an issue with the ['Install Sudo in container' step](https://dev.azure.com/Lightning-AI/Tutorials/_build/results?buildId=177022&view=logs&jobId=92a27946-4b27-5f21-4be8-dbadb7970d16&j=92a27946-4b27-5f21-4be8-dbadb7970d16&t=c79061c9-906c-5adf-0fd6-f111586a53ac&l=12)

**Review by Borda — APPROVED** [review]

**Comment by speediedan** [comment]
LOVE this [new workflow](https://github.com/Lightning-AI/lightning/pull/18783/files) to keep tutorials updated! YOU'RE AWEOME Jirka! 

Any chance we can get this FTS update PR merged prior to the automated tutorial update that is scheduled to run this Sunday at midnight? I noticed there's a layout issue with the current version that would be addressed by the automated update so the new workflow is already bearing fruit!!! Thanks again!
@Borda @carmocca

**Review by Borda — APPROVED** [review]

**Comment by speediedan** [comment]
awesome :rocket:, thanks!
